contracts/rewards/src/state.rs

```diff
@@ -42,9 +42,10 @@ impl EpochTally {
         }
     }
 
+    /// IMPORTANT: worker must be a valid address
     pub fn record_participation(mut self, worker: Addr) -> Self {
         self.participation
-            .entry(worker)
+            .entry(worker.to_string())
```

Comment:
If it's a must, I'd rather adding a check to this method that panics if the address isn't valid.

Reply:
It's a little clunky to validate that here, we need access to `Api` somehow: https://docs.rs/cosmwasm-std/latest/cosmwasm_std/trait.Api.html